src/foucautSinusoidal.js

```diff
@@ -0,0 +1,47 @@
+import {geoProjectionMutator as projectionMutator} from "d3-geo";
+import {cos, halfPi, pi, sin, sqrt} from "./math";
+import {solve} from "./newton";
+
+export function foucautSinusoidalRaw(alpha) {
+  var beta = 1 - alpha;
+  var equatorial = raw(pi, 0)[0] - raw(-pi, 0)[0],
+      polar = raw(0, halfPi)[1] - raw(0, -halfPi)[1],
+      ratio = sqrt(2 * polar / equatorial);
+
+  function raw(lambda, phi) {
+    var cosphi = cos(phi),
+        sinphi = sin(phi);
+    return [
+      cosphi / (beta + alpha * cosphi) * lambda,
+      beta * phi + alpha * sinphi
+    ];
+  }
+
+  function forward(lambda, phi) {
+    var p = raw(lambda, phi);
+    return [ p[0] * ratio, p[1] / ratio ];
+  }
+
+  forward.invert = function(x, y) {
+    var phi = solve(function(phi) {
+      return forward(0, phi)[1];
+    }, y);
+    var lambda = x / ratio * (alpha + beta / cos(phi));
```

Comment:
Since the function you’re passing to *solve* doesn’t depend on *x* or *y*, it’d be better to move it outside of *forward*.invert (so as to avoid creating a new closure every time *forward*.invert is called).